### ISSUE TITLE
Duplicate Registered Metric with Exporter Helper and OTel Metrics

### ISSUE THREAD

**Issue by cpheps** [open-telemetry/opentelemetry-collector#9514]
**Describe the bug**
This occurs when the `useOtelForInternalMetrics` flag is enabled in pre-v0.93.0 versions and v0.93.0. It occurs when the collector is used as a library and a collector service instance is stoped and restarted with the same config without restarting the process. The config must use an exporter that uses the exporter helper and sending queue enabled. The following error occurs when curling the `http://localhost:8888/metrics` endpoint:

```shell
=> curl http://localhost:8888/metrics
An error has occurred while serving metrics:

collected metric "otelcol_exporter_queue_size" { label:{name:"exporter" value:"elasticsearch/es__logs"} label:{name:"service_instance_id" value:"681b6c18-7c4b-406b-9cf0-ba7923084785"} label:{name:"service_name" value:"dist/collector_darwin_arm64"} label:{name:"service_version" value:"v1.44.0"} gauge:{value:0}} was collected before with the same name and label values
```

If the offending exporter is removed and the config is reloaded, all without restarting the process the metrics endpoint works again.

**Steps to reproduce**
1. Use collector as a library
2. Start a Service instance with a config using the OTLP exporter with the `sending_queue` enabled.
3. Curl the internal metrics endpoint to observe it works
4. Stop the Service and start a new instance with the same config.
5. Curl the internal metrics endpoint and observe the error

**What did you expect to see?**
The internal metrics endpoint working as it does with `useOtelForInternalMetrics` disabled.

**What did you see instead?**
Error on internal metrics endpoint

**What version did you use?**
v0.92.0

**What config did you use?**
```yaml
receivers:
    hostmetrics:
        scrapers:
            load:
                metrics: null
exporters:
    otlp:
        endpoint: localhost:4317
        retry_on_failure:
            enabled: true
        sending_queue:
            enabled: true
service:
    pipelines:
        metrics:
            receivers:
                - hostmetrics
            exporters:
                - otlp
```

**Environment**
OS: macOS 14.2.1 (23C71)
Compiler(if manually compiled): go 1.21.6 darwin/arm64

**Comment by cpheps** [comment]
Looking at this a bit more I noticed that the `queue_size` metric I think is being created using [meter.Int64ObservableGuage](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.92.0/exporter/exporterhelper/queue_sender.go#L151) where most other metrics are being created via the `go.opencensus.io/stats/view` package `Register` function. I'm not sure if this is related or not.

**Comment by Symbianx** [comment]
I just bumped into this after updating to 0.93.0 using the `loadbalancing` exporter configured for `otlp` using the following config:
```yaml
receivers:
  otlp:
    protocols:
      grpc:
      http:
  jaeger:
    protocols:
      thrift_http:
processors:
  batch:
  memory_limiter:
    limit_percentage: 80
    spike_limit_percentage: 25
    check_interval: 5s
extensions:
  memory_ballast:
    # Memory Ballast size should be max 1/3 to 1/2 of memory.
    size_in_percentage: 40
  health_check:
exporters:
  loadbalancing:
    protocol:
      otlp:
        tls:
          insecure: true
        sending_queue:
        retry_on_failure:
          enabled: true
    resolver:
      dns:
        hostname: "${OTEL_COLLECTOR_HOSTNAME}"
        port: 4317
service:
  extensions: [ memory_ballast, health_check ]
  telemetry:
    logs:
      encoding: json
  pipelines:
    traces/1:
      receivers: [ otlp, jaeger ]
      processors: [ memory_limiter, batch ]
      exporters: [ loadbalancing ]
```

**Comment by cpheps** [comment]
@codeboten Do you have any insight? Looks like you've been involved in this feature flag a bit. It's not quite obvious to me why this one metric causes the issue with the new OTel internal metrics.

**Comment by cpheps** [comment]
I'm wondering if it's related to this PR https://github.com/open-telemetry/opentelemetry-go/pull/4820. I noticed the duplicate check doesn't exist in the version of the opentelemetry-go library. I'll retry this with OTel v0.94.1 as it does contain this fix.

**Comment by cpheps** [comment]
Confirmed that this has been fixed in OTel v0.94.1. Closing this issue.
